docs/README.md

```diff
@@ -178,25 +178,25 @@
     * [Health Check Library](extensions/healthcheck-library.md)
   * [CA Rotation in Extensions](extensions/ca-rotation.md)
   * Blob storage providers
-    * [`BackupBucket` resource](extensions/backupbucket.md)
-    * [`BackupEntry` resource](extensions/backupentry.md)
+    * [`BackupBucket` resource](extensions/resources/backupbucket.md)
+    * [`BackupEntry` resource](extensions/resources/backupentry.md)
   * DNS providers
     * [`DNSRecord` resources](extensions/dnsrecord.md)
   * IaaS/Cloud providers
     * [Control plane customization webhooks](extensions/controlplane-webhooks.md)
-    * [`Bastion` resource](extensions/bastion.md)
-    * [`ControlPlane` resource](extensions/controlplane.md)
-    * [`ControlPlane` exposure resource](extensions/controlplane-exposure.md)
-    * [`Infrastructure` resource](extensions/infrastructure.md)
-    * [`Worker` resource](extensions/worker.md)
+    * [`Bastion` resource](extensions/resources/bastion.md)
+    * [`ControlPlane` resource](extensions/resources/controlplane.md)
+    * [`ControlPlane` exposure resource](extensions/resources/controlplane-exposure.md)
+    * [`Infrastructure` resource](extensions/resources/infrastructure.md)
+    * [`Worker` resource](extensions/resources/worker.md)
   * Network plugin providers
-    * [`Network` resource](extensions/network.md)
+    * [`Network` resource](extensions/resources/network.md)
   * Operating systems
-    * [`OperatingSystemConfig` resource](extensions/operatingsystemconfig.md)
+    * [`OperatingSystemConfig` resource](extensions/resources/operatingsystemconfig.md)
   * Container runtimes
-    * [`ContainerRuntime` resource](extensions/containerruntime.md)
+    * [`ContainerRuntime` resource](extensions/resources/containerruntime.md)
   * Generic (non-essential) extensions
-    * [`Extension` resource](extensions/extension.md)
+    * [`Extension` resource](extensions/resources/extension.md)
```

Comment:
Please, for each moved file take care to find the usages outside of this repo and raise a PR for them.

For example the extension.md file is referenced by repos like [registry-cache](https://github.com/gardener/gardener-extension-registry-cache/blob/2f99e1fa965c8b8d33778fc82861889ba1e9a33f/docs/development/extension-registry-cache.md?plain=1#L18).